src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImplGetUACTest.java

```diff
@@ -0,0 +1,174 @@
+package uk.gov.ons.ctp.integration.contactcentresvc.service.impl;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static uk.gov.ons.ctp.integration.contactcentresvc.CaseServiceFixture.AN_AGENT_ID;
+import static uk.gov.ons.ctp.integration.contactcentresvc.CaseServiceFixture.A_QUESTIONNAIRE_ID;
+import static uk.gov.ons.ctp.integration.contactcentresvc.CaseServiceFixture.A_REGION;
+import static uk.gov.ons.ctp.integration.contactcentresvc.CaseServiceFixture.A_UAC;
+import static uk.gov.ons.ctp.integration.contactcentresvc.CaseServiceFixture.UUID_0;
+
+import java.util.Optional;
+import java.util.UUID;
+import lombok.SneakyThrows;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.server.ResponseStatusException;
+import uk.gov.ons.ctp.common.domain.FormType;
+import uk.gov.ons.ctp.common.error.CTPException;
+import uk.gov.ons.ctp.common.error.CTPException.Fault;
+import uk.gov.ons.ctp.integration.caseapiclient.caseservice.model.CaseContainerDTO;
+import uk.gov.ons.ctp.integration.caseapiclient.caseservice.model.SingleUseQuestionnaireIdDTO;
+import uk.gov.ons.ctp.integration.contactcentresvc.cloud.CachedCase;
+import uk.gov.ons.ctp.integration.contactcentresvc.representation.UACRequestDTO;
+import uk.gov.ons.ctp.integration.contactcentresvc.representation.UACResponseDTO;
+import uk.gov.ons.ctp.integration.contactcentresvc.service.CaseService;
+
+/** Unit Test {@link CaseService#getUACForCaseId(UUID, UACRequestDTO) getUACForCaseId}. */
+@RunWith(MockitoJUnitRunner.class)
+public class CaseServiceImplGetUACTest extends CaseServiceImplTestBase {
+
+  @Test
+  public void testGetUACCECase() throws Exception {
+    doGetUACTest("CE", false);
+  }
+
+  @Test
+  public void testGetUACCECaseForIndividual() throws Exception {
+    doGetUACTest("CE", true);
+  }
+
+  @Test
+  public void testGetUACHHCase() throws Exception {
+    doGetUACTest("HH", false);
+  }
+
+  @Test
+  public void testGetUACHHCaseForIndividual() throws Exception {
+    doGetUACTest("HH", true);
+  }
+
+  @Test
+  public void testGetUACSPGCase() throws Exception {
+    doGetUACTest("SPG", false);
+  }
+
+  @Test
+  public void testGetUACSPGCaseForIndividual() throws Exception {
+    doGetUACTest("SPG", true);
+  }
+
+  @Test
+  public void testGetUACHICase() throws Exception {
```

Comment:
Doesn't throw an Exception as you've caught it

Reply:
Yep, that error really bugs me, changed.